spec/core/connector_settings_spec.rb

```diff
@@ -62,6 +62,153 @@
     end
   end
 
+  context '.filtering' do
+    context 'filtering is present' do
+      let(:elasticsearch_response) {
+        {
+          :filtering => [
+            {
+              :domain => 'DEFAULT',
+              :active => {
+                :active_rules => [],
+                :advanced_config => {},
+              }
+            }
+          ]
+        }
+      }
+
+      it 'extracts filtering field' do
+        filtering = subject.filtering
+        first_filter = filtering[0]
+
+        expect(first_filter[:domain]).to eq('DEFAULT')
+        expect(first_filter[:active][:active_rules]).to_not be_nil
+        expect(first_filter[:active][:advanced_config]).to_not be_nil
+      end
+    end
+
+    context 'filtering is not present' do
+      it 'returns default filtering object' do
+        filtering = subject.filtering
+
+        expect(filtering).to_not be_nil
+        expect(filtering.empty?).to eq(true)
+      end
+    end
+  end
+
+  context '.active_advanced_filter_config' do
+    shared_examples_for 'extracts active_filter_config' do
+      it 'extracts active_advanced_filter_config' do
+        advanced_config = subject.active_filter_config
+
+        expect(advanced_config).to_not be_nil
+        expect(advanced_config[:find][:options][:skip]).to eq(10)
+      end
+    end
+
+    context 'active advanced config exists as one element array' do
+      let(:elasticsearch_response) {
+        {
+          :filtering => [
+            {
+              :domain => 'DEFAULT',
+              :active => {
+                :active_rules => [],
+                :advanced_config => {
+                  :find => {
+                    :options => {
+                      :skip => 10
+                    }
+                  }
+                },
+              },
+              :draft => {
+                :active_rules => [],
+                :advanced_config => {}
+              }
+            },
+          ]
+        }
+      }
+
+      it_behaves_like 'extracts active_filter_config'
+    end
+
+    context 'active advanced config exists as an object' do
+      let(:elasticsearch_response) {
+        {
+          :filtering => {
+              :domain => 'DEFAULT',
+              :active => {
+                :active_rules => [],
+                :advanced_config => {
+                  :find => {
+                    :options => {
+                      :skip => 10
+                    }
+                  }
+                },
+              },
+              :draft => {
+                :active_rules => [],
+                :advanced_config => {}
+              }
+            }
+        }
+      }
+
+      it_behaves_like 'extracts active_filter_config'
```

Comment:
praise: I'm impressed that you've picked up using shared examples. This is a great tool that we underutilize, IMO